juju/client/connector.py

```diff
@@ -0,0 +1,121 @@
+import asyncio
+import logging
+import macaroonbakery.httpbakery as httpbakery
+
+from juju.client.connection import Connection
+from juju.client.jujudata import JujuData
+from juju.errors import JujuConnectionError, JujuError
+
+log = logging.getLogger('connector')
+
+
+class Connector:
+    def __init__(self, loop=None, max_frame_size=None, bakery_client=None):
+        self.max_frame_size = max_frame_size
+        self.loop = loop or asyncio.get_event_loop()
+        self.bakery_client = bakery_client
+        self._connection = None
+        self.controller_name = None
+        self.model_name = None
+        self.jujudata = JujuData()
+
+    def is_connected(self):
+        return self._connection is not None
+
+    def connection(self):
+        if not self.is_connected():
+            raise Exception('not connected')
+        return self._connection
+
+    async def connect(self, *args, **kw):
+        """Connect to an arbitrary Juju model.
+
+        args and kw are passed through to Connection.connect()
+
+        """
+        kw.setdefault('loop', self.loop)
+        kw.setdefault('max_frame_size', self.max_frame_size)
+        kw.setdefault('bakery_client', self.bakery_client)
+        self._connection = await Connection.connect(*args, **kw)
+
+    async def disconnect(self):
+        """Shut down the watcher task and close websockets.
+        """
+        if self._connection:
+            log.debug('Closing model connection')
+            await self._connection.close()
+            self._connection = None
+
+    async def connect_controller(self, controller_name):
+        """Connect to a controller by name. If the name is empty, it
+        connect to the current controller.
```

Comment:
So make controller_name optional? Also s/controller_name/name maybe.

Reply:
+1 to optional `name`

Reply:
done